### PR TITLE
Clarify LTS lifetime

### PR DESCRIPTION
**Pull request by yanesca**
## Description

Users can get more value out of our LTS releases and plan transition between versions better if the documentation is more clear about the lifetime of LTS branches.

## Gatekeeper checklist

- [x] **changelog** not required
- [ ] **backport** will be provided after approval
- [x] **tests** not required

### CODE REVIEW

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM

File: BRANCHES.md [hunk 1]
```diff
@@ -23,6 +23,9 @@ the API of 3.(x+1) is backward compatible with 3.x). We only break API
 compatibility on major version changes (e.g. from 3.x to 4.0). We also maintain
 ABI compatibility within LTS branches; see the next section for details.
 
+New LTS branches are usually created around the time of a new major version.
+After a new LTS branch is created, it remains supported for at least 3 years.
```
Comment (daverodgman) [inline BRANCHES.md:27]:
```suggestion
When a new LTS branch is created, it usually remains supported for three years.
```

**Review by daverodgman — CHANGES_REQUESTED** [review]
It's not a given that an LTS will be supported for 3 years; I've suggested some more precise wording.

**Review by daverodgman — DISMISSED** [review]

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Pull request merged**